libs/qCC_io/FileIOFilter.cpp

```diff
@@ -455,12 +455,17 @@ bool FileIOFilter::CheckForSpecialChars(const QString& filename)
 	return (filename.normalized(QString::NormalizationForm_D) != filename);
 }
 
-bool FileIOFilter::HandleGlobalShift(const CCVector3d& P, CCVector3d& Pshift, LoadParameters& loadParameters, bool useInputCoordinatesShiftIfPossible/*=false*/)
+bool FileIOFilter::HandleGlobalShift(	const CCVector3d& P,
+										CCVector3d& Pshift,
+										bool& preserveCoordinateShift,
```

Comment:
...and I think the logic is incorrect anyways.

The first time this is called, `preserveCoordinateShift` is true, the dialog comes up, and the user turns it off. `preserveCoordinateShift` gets set to false and `loadParameters.preserveShiftOnSave` gets set to false.

The next time through, `shiftAlreadyEnabled` is true, `preserveCoordinateShift` gets set to false, but then in `ccGlobalShiftManager::Handle` it gets set to true again which is incorrect.

The result is the first cloud has the shift turned off, but the rest (apply all) have it turned on.

Is there a way to simplify this whole thing?

Reply:
1) Well, I simply 'mimicked' the way 'coordinatesShift' was transmitted (and I didn't want to miss any call to ccGlobalShiftManager::Handle so I chose the 'change of signature' way to force me and other developers to take this parameter into account).

Reply:
2) In ccGlobalShiftManager::Handle it shouldn't be necessarily set to true (we actually use 's_lastInfo.preserve' for the default state of the checkbox... I know it's a bit complicated, There have been too many iterations, we should probably restart from scratch, but it will be a very sensitive and tedious work :D).

Reply:
In practice it is not working though, so whatever is supposed to be resetting it to false in ccGlobalShiftManager::Handle isn't.

As I pointed out - the first one gets it turned off properly and subsequent ones have it on. I'm using an E57 file with several point clouds in it.

Reply:
3) I still need to test everything anyway (it may take some time!)